scripts/kwarg_permissions.py

```diff
@@ -0,0 +1,145 @@
+from typing import Optional
+
+import libcst as cst
+import libcst.codemod.visitors as codevisitors
+import libcst.matchers as m
+from libcst import codemod
+
+from disnake import Permissions
+
+
+class PermissionTypings(codemod.VisitorBasedCodemodCommand):
+    DESCRIPTION: str = "Adds overloads for all permissions."
+
+    def __init__(self, context: codemod.CodemodContext):
+        super().__init__(context)
+        # add all of the permissions
+        self.permissions = sorted(Permissions.VALID_FLAGS.keys())
+
+    def transform_module(self, tree: cst.Module) -> cst.Module:
+        if "@_overload_with_permissions" not in tree.code:
```

Comment:
Is it possible to completely skip parsing of files that don't contain this string? It's not really worth it to parse the module into a CST (certainly the most expensive part of the process), only to skip it based on the original raw text contents.

If not possible with `libcst.tool`, we might be able to write our own entrypoint that filters files before passing them into libcst, shouldn't be too complex since they provide a couple useful tools: https://libcst.readthedocs.io/en/latest/codemods.html#command-line-toolkit

Alternatively, we could use the nox session to filter them before calling libcst; it would require always using nox instead of calling libcst directly, but that should be fine.

Reply:
Not easily, unfortunately. We can do this if we write our own entrypoint, which we could do in this pr or a different one.

Reply:
#713